src/index/testing.rs

```diff
@@ -128,7 +128,20 @@ impl Context {
     let balances = balances.as_mut();
     balances.sort_by_key(|(outpoint, _)| *outpoint);
 
-    for (_, balances) in balances.iter_mut() {
+    let mut burned_balances = 0;
+
+    for (outpoint, balances) in balances.iter_mut() {
+      if self
+        .index
+        .get_output_script(outpoint)
+        .ok()
+        .is_some_and(|script| script.is_op_return())
+      {
+        for (_, balance) in balances.iter() {
+          burned_balances += *balance;
+        }
+      }
+
```

Comment:
To be extra sure we should not only calculate the burned amount here but also ask `self.index` for the burned amount.